rustls/src/client/tls12.rs

```diff
@@ -1047,12 +1047,13 @@ impl<C: CryptoProvider> State<ClientConnectionData> for ExpectFinished<C> {
         // Constant-time verification of this is relatively unimportant: they only
         // get one chance.  But it can't hurt.
         let _fin_verified =
-            constant_time::verify_slices_are_equal(&expect_verify_data, &finished.0)
-                .map_err(|_| {
-                    cx.common
-                        .send_fatal_alert(AlertDescription::DecryptError, Error::DecryptError)
-                })
-                .map(|_| verify::FinishedMessageVerified::assertion())?;
+            if subtle::ConstantTimeEq::ct_eq(&expect_verify_data[..], &finished.0).into() {
```

Comment:
I'd import `ConstantTimeEq` directly, and probably write this as a `match` expression. Or better, maybe this is a helper wrapper around `CommonState::send_fatal_alert()`?